io/io/src/TKey.cxx

```diff
@@ -741,6 +741,20 @@ TObject *TKey::ReadObj()
       return (TObject*)ReadObjectAny(0);
    }
 
+   struct CleanUp {
+      TKey *fKey;
+      CleanUp(TKey *key) : fKey(key) {}
+      ~CleanUp() {
+         delete fKey->fBufferRef;
+         fKey->fBufferRef = nullptr;
+         // This 'if' is why we can't use TKey::DeleteBuffer
+         if (fKey->fObjlen > fKey->fNbytes - fKey->fKeylen)
+            delete [] fKey->fBuffer;
+         fKey->fBuffer = nullptr;
```

Comment:
Could you add a comment why the buffer shall be zeroed even if the buffer isn't deleted, i.e. who owns the memory `fBuffer` was pointing to?